tools/clang/test/SemaHLSL/attributes.hlsl

```diff
@@ -350,11 +350,11 @@ int uav() {
   return i;
 }
 
-[domain] int domain_fn_missing() { return 1; }          // expected-error {{'domain' attribute takes one argument}} fxc-pass {{}}
+[domain] [shader("pixel")]int domain_fn_missing() { return 1; }          // expected-error {{'domain' attribute takes one argument}} fxc-pass {{}}
```

Comment:
Why `[shader("pixel")]`?  That would be incorrect for this shader stage.  I would expect we would want to emit some error in the future for a case like this.  On top of that, with the addition of shader attributes and a lib target, we might do additional attribute diagnostics that these sorts of tests are not prepared for.  It would be more work to make these error-free if we were doing the diagnostics in Sema that we would want to do.

I think a better option is to just add the expected-warning message to these, rather than add shader attributes.

If the issue is the addition of the lib target run line and this test behaves differently in that case, then we should split the test into lib/non-lib files.

Reply:
I've added warnings instead of shader attributes.